libs/features/personalization/personalization.js

```diff
@@ -1124,6 +1128,104 @@ async function callMartech(config) {
   }
   return targetManifests;
 }
+
+function roundToQuarter(num) {
+  return Math.ceil(num / 250) / 4;
+}
+
+function calculateResponseTime(responseStart) {
+  const responseTime = Date.now() - responseStart;
+  return roundToQuarter(responseTime);
+}
+
+function sendTargetResponseAnalytics(failure, responseStart, timeoutLocal, message) {
+  // temporary solution until we can decide on a better timeout value
+  const responseTime = calculateResponseTime(responseStart);
+  const timeoutTime = roundToQuarter(timeoutLocal);
+  let val = `target response time ${responseTime}:timed out ${failure}:timeout ${timeoutTime}`;
+  if (message) val += `:${message}`;
+  // eslint-disable-next-line no-underscore-dangle
+  window._satellite?.track?.('event', {
+    documentUnloading: true,
+    xdm: {
+      eventType: 'web.webinteraction.linkClicks',
+      web: {
+        webInteraction: {
+          linkClicks: { value: 1 },
+          type: 'other',
+          name: val,
+        },
+      },
+    },
+    data: { _adobe_corpnew: { digitalData: { primaryEvent: { eventInfo: { eventName: val } } } } },
+  });
+}
+
+const handleAlloyResponse = (response) => {
+  const items = (
+    (response.propositions?.length && response.propositions)
+    || (response.decisions?.length && response.decisions)
+    || []
+  ).map((i) => i.items).flat();
+
+  if (!items?.length) return [];
```

Comment:
This is unecessary. You can just continue the chain since `[].map(f) === (null)?.map(f) ?? [] === []`

```javascript

const handleAlloyResponse = (response) => ((response.propositions || response.decisions))
    ?.map((i) => i.items)
    ?.flat()
    ?.map(() => /*more stuff */)
    ?.filter(Boolean) ?? [];

```